### PR TITLE
[Opaque pointer] Port tests in AMDGPU/NVPTX/SYCL-FUSION etc to opaque pointers

### PR DESCRIPTION
**Pull request by jsji**
This is in preparation of -DINTEL_SYCL_OPAQUEPOINTER_READY=1 

- Port barrier_intrinsic.ll to opaque pointer
- Port spirv-to-ir-wrapper.ll to opaque pointers
- Update NVPTX tests to opaque pointers
- Update AMDGPU tests to opaque pointers
- Update sycl fusion tests to opaque pointers

### CODE REVIEW

File: sycl-fusion/test/internalization/promote-local-vec.ll [hunk 1]
```diff
@@ -30,59 +30,59 @@ declare spir_func void @__itt_offload_wi_start_wrapper() #3
...
 ; CHECK:  entry:
-; CHECK:    [[KERNELONE_ACCTMP31_SROA_0_0__SROA_IDX:%.*]] = getelementptr inbounds [[TYPE1]], %1* [[KERNELONE_ACCTMP3]], i64 0, i32 0, i32 0, i64 0
-; CHECK:    [[KERNELONE_ACCTMP31_SROA_0_0_COPYLOAD:%.*]] = load i64, i64* [[KERNELONE_ACCTMP31_SROA_0_0__SROA_IDX]], align 1
+; CHECK:    [[KERNELONE_ACCTMP31_SROA_0_0__SROA_IDX:%.*]] = getelementptr inbounds [[TYPE1]], ptr [[KERNELONE_ACCTMP3]], i64 0, i32 0, i32 0, i64 0
+; CHECK:    [[KERNELONE_ACCTMP31_SROA_0_0_COPYLOAD:%.*]] = load i64, ptr [[KERNELONE_ACCTMP31_SROA_0_0__SROA_IDX]], align 1
 ; CHECK:    [[TMP0:%.*]] = urem i64 [[KERNELONE_ACCTMP31_SROA_0_0_COPYLOAD]], 16
-; CHECK:    [[ADD_PTR_I_I:%.*]] = getelementptr inbounds [[TYPE0]], [[TYPE0]] addrspace(3)* [[KERNELONE_ACCTMP]], i64 [[TMP0]]
+; CHECK:    [[ADD_PTR_I_I:%.*]] = getelementptr inbounds [[TMP02:%.*]], ptr addrspace(3) [[KERNELONE_ACCTMP]], i64 [[TMP0]]
```
Comment (sommerlukas) [inline sycl-fusion/test/internalization/promote-local-vec.ll:50]: Were these new checks created manually or by an LLVM script? If they were created manually, can we change `TMP02` to `TYPE...` to indicate that this is a type?

**Review by sommerlukas — COMMENTED** [review]
Only reviewed the files relating to kernel fusion (`sycl-fusion/test`), just a minor nit. 

Thanks for taking care of this.

File: sycl-fusion/test/internalization/promote-private-vec.ll [hunk 2]
```diff
@@ -30,10 +30,10 @@ declare spir_func void @__itt_offload_wi_start_wrapper() #3
...
 
 ; Function Attrs: noinline nounwind
-declare  spir_func void @__itt_offload_wi_finish_stub(i64 addrspace(4)* %group_id, i64 %wi_id) #4
+declare  spir_func void @__itt_offload_wi_finish_stub(ptr addrspace(4) %group_id, i64 %wi_id) #4
 
 ; Function Attrs: noinline nounwind
-declare spir_func void @__itt_offload_wi_start_stub(i64 addrspace(4)* %group_id, i64 %wi_id, i32 %wg_size) #4
+declare spir_func void @__itt_offload_wi_start_stub(ptr addrspace(4) %group_id, i64 %wi_id, i32 %wg_size) #4
 
 define spir_kernel void @fused_0(%0 addrspace(1)* align 16 %KernelOne_accTmp, %1* byval(%1) align 8 %KernelOne_accTmp3, %0 addrspace(1)* align 16 %KernelOne_accIn1, %1* byval(%1) align 8 %KernelOne_accIn16, %0 addrspace(1)* align 16 %KernelOne_accIn2, %0 addrspace(1)* align 16 %KernelTwo_accOut, %0 addrspace(1)* align 16 %KernelTwo_accIn3) !kernel_arg_addr_space !12 !kernel_arg_access_qual !13 !kernel_arg_type !14 !kernel_arg_type_qual !15 !kernel_arg_base_type !14 !kernel_arg_name !16 !sycl.kernel.promote !17 !sycl.kernel.promote.localsize !18 {
```
Comment (sommerlukas) [inline sycl-fusion/test/internalization/promote-private-vec.ll:38]: Also, is there any reason the typed pointers in this function were not replaced with opaque pointers?

**Review by MrSidims — APPROVED** [review]
spirv-to-ir-wrapper.ll is LGTM

File: sycl-fusion/test/internalization/promote-local-vec.ll [hunk 3]
```diff
@@ -30,59 +30,59 @@ declare spir_func void @__itt_offload_wi_start_wrapper() #3
...
 
-define spir_kernel void @fused_0(%0 addrspace(1)* align 16 %KernelOne_accTmp, %1* byval(%1) align 8 %KernelOne_accTmp3, %0 addrspace(1)* align 16 %KernelOne_accIn1, %1* byval(%1) align 8 %KernelOne_accIn16, %0 addrspace(1)* align 16 %KernelOne_accIn2, %0 addrspace(1)* align 16 %KernelTwo_accOut, %0 addrspace(1)* align 16 %KernelTwo_accIn3) !kernel_arg_addr_space !12 !kernel_arg_access_qual !13 !kernel_arg_type !14 !kernel_arg_type_qual !15 !kernel_arg_base_type !14 !kernel_arg_name !16 !sycl.kernel.promote !17 !sycl.kernel.promote.localsize !18 {
+define spir_kernel void @fused_0(%0 addrspace(1)* align 16 %KernelOne_accTmp, ptr byval(%1) align 8 %KernelOne_accTmp3, %0 addrspace(1)* align 16 %KernelOne_accIn1, ptr byval(%1) align 8 %KernelOne_accIn16, %0 addrspace(1)* align 16 %KernelOne_accIn2, %0 addrspace(1)* align 16 %KernelTwo_accOut, %0 addrspace(1)* align 16 %KernelTwo_accIn3) !kernel_arg_addr_space !12 !kernel_arg_access_qual !13 !kernel_arg_type !14 !kernel_arg_type_qual !15 !kernel_arg_base_type !14 !kernel_arg_name !16 !sycl.kernel.promote !17 !sycl.kernel.promote.localsize !18 {
```
Comment (victor-eds) [inline sycl-fusion/test/internalization/promote-local-vec.ll:38]: Typed pointers still used here and in metadata at the bottom of the file

File: sycl-fusion/test/syclcp/syclcp.ll [hunk 4]
```diff
@@ -45,134 +45,134 @@ declare spir_func void @__itt_offload_wi_start_wrapper() #4
...
 
 ; Function Attrs: noinline nounwind
-declare spir_func void @__itt_offload_wi_start_stub(i64 addrspace(4)* %group_id, i64 %wi_id, i32 %wg_size) #5
+declare spir_func void @__itt_offload_wi_start_stub(ptr addrspace(4) %group_id, i64 %wi_id, i32 %wg_size) #5
 
 
-define spir_kernel void @fused_0(%0* byval(%0) align 8 %KernelOne_accTmp3, i32 addrspace(1)* align 4 %KernelOne_accIn1, %0* byval(%0) align 8 %KernelOne_accIn16, i32 addrspace(1)* align 4 %KernelOne_accIn2, i32 addrspace(1)* align 4 %KernelTwo_out, %2* byval(%2) align 4 %KernelTwo_coef) !kernel_arg_addr_space !16 !kernel_arg_access_qual !7 !kernel_arg_type !17 !kernel_arg_type_qual !9 !kernel_arg_base_type !17 !kernel_arg_name !18 !sycl.kernel.constants !19 {
+define spir_kernel void @fused_0(ptr byval(%0) align 8 %KernelOne_accTmp3, ptr addrspace(1) align 4 %KernelOne_accIn1, ptr byval(%0) align 8 %KernelOne_accIn16, ptr addrspace(1) align 4 %KernelOne_accIn2, ptr addrspace(1) align 4 %KernelTwo_out, %2* byval(%2) align 4 %KernelTwo_coef) !kernel_arg_addr_space !16 !kernel_arg_access_qual !7 !kernel_arg_type !17 !kernel_arg_type_qual !9 !kernel_arg_base_type !17 !kernel_arg_name !18 !sycl.kernel.constants !19 {
```
Comment (victor-eds) [inline sycl-fusion/test/syclcp/syclcp.ll:54]: Mised a typed pointer here

**Review by victor-eds — COMMENTED** [review]
Thanks for taking care of this. In addition to the cases @sommerlukas and I point out, there are some missing typed pointers in tests unmodified by this PR under `sycl-fusion/test/kernel-fusion`.

File: sycl-fusion/test/internalization/promote-local-vec.ll (hunk 1 above)
Comment (jsji) [reply sycl-fusion/test/internalization/promote-local-vec.ll]: Done.

**Comment by jsji** [comment]
> Thanks for taking care of this. In addition to the cases @sommerlukas and I point out, there are some missing typed pointers in tests unmodified by this PR under `sycl-fusion/test/kernel-fusion`.

Good catch, done.

File: sycl-fusion/test/syclcp/syclcp.ll (hunk 4 above)
Comment (jsji) [reply sycl-fusion/test/syclcp/syclcp.ll]: Good catch, done.

File: sycl-fusion/test/internalization/promote-local-vec.ll (hunk 3 above)
Comment (jsji) [reply sycl-fusion/test/internalization/promote-local-vec.ll]: Good catch, done.

**Review by victor-eds — APPROVED** [review]
Kernel fusion part LGTM

**Review by sommerlukas — APPROVED** [review]
Tests relating to kernel fusion LGTM now.

**Comment by jsji** [comment]
@intel/llvm-gatekeepers Can we get this merged? Thanks!

**Pull request merged**